design/one-pager-managed-resource-api-design.md

```diff
@@ -0,0 +1,398 @@
+
+The decision for an external resource name is made by the controller of that managed resource in `Create` phase.
+Possible cases are as following:
+* The provider doesn't allow us to specify a name. Fetch it after its client's `POST` call. Override `crossplane.io/external-name`
+annotation's value with what you get as name.
+* The provider allows to specify a name.
+  * If `crossplane.io/external-name` annotation has a value use it. If it's empty;
+  * Use managed resource's name and write that into `crossplane.io/external-name` annotation.
+
+Use the value of that annotation as external resource name in _all_ queries.
+
+#### Field Names
+
+The thinking process of naming decision could respect provider's decision as the starting point and deviate from that
+if there is a very good reason to. Naming parity makes inter-resource references easier and lowers the entry barrier for
+both developers and users. This is especially handy in cases where Crossplane does not yet support the _referred_ resource
+or user may not want some of their resources in their Crossplane environment.
+
+In some cases, we might have collisions or cases where provider field name is too similar to another field in the
+managed resource's Custom Resource fields. The section [Owner-based Struct Tree](#owner-based-struct-tree) tries to
+tackle that.
+
+### High Fidelity
+
+Related to https://github.com/crossplaneio/crossplane/issues/621 and https://github.com/crossplaneio/crossplane/issues/530
+
+Crossplane managed resources should expose everything that provider exposes to its users as much as possible. A few 
+benefits of that high fidelity:
+
+* The ability to read and set every 'knob' the external resource supports. This would make it easy to support wide 
+variety of customizations without having to think of them in development time.
+* An obvious mapping of Crossplane's fields to cloud provider API fields, exact name match as much as possible. This 
+would make it easier to work with Crossplane for users who are familiar with the given provider in many ways 
+including easier troubleshooting and cooperation with other existing infrastructure tools.
+
+What goes into `Spec`:
+* Does the provider allow configuration of this parameter at _any_ point in the life cycle of the resource? Include if yes.
+  This includes the fields that are late-initialized, meaning some fields could be populated by the provider when you
+  actually create the resource but it also gives you the ability to change them. An example for this could be
+  auto-generated resource tags or resource-specific defaults. If the provider tags the resource without us telling them to
+  do so, controller should update `Spec` and input user should make changes on that current value of the field.
+  Related to https://github.com/crossplaneio/crossplane-runtime/issues/10
+
+Note that you should make updates to `Spec` fields that are empty. We do not override user's desired state and if they
```

Comment:
> Note that you should make updates to `Spec` fields that are empty.

Does this mean that controllers should only make updates to `Spec` fields that are empty, i.e. late initialization? It's a little ambiguous who "you" is.